### DISCUSSION TITLE
AnyOf Matching

### DISCUSSION THREAD

**Discussion by thangurion** [AngryBeaver/beavers-crafting#35] [General]
Hi, is it possible to fill more than one argument item.system.source and to look in the any of item for one of them?
I´ve attached some screenshots. I hope you can understand what i try to make. 
In the "anyof-item" i need an operator that search the noted string in "the item.system.source" field. Or are wildcards possible?

![ANYOF](https://user-images.githubusercontent.com/113504992/218827295-7e61a2bd-e29b-4774-81aa-ea6dce62ff5e.png)
![INGREDIENT](https://user-images.githubusercontent.com/113504992/218827301-0c2c69fe-9c29-4516-962d-3da3739810b7.png)

Thank you for this great module. My Elven Druidess will be very happy about the new possibilities.

**Comment by AngryBeaver** [comment]
well it is a macro you can javascript what ever you see fit.
so yes you can do so with standard javascript, nothing to do with my module :-)

e.g. something like:

````
const systemSourceParts =  item.system.source.split("|");
let found = false;
for(const systemSourcePart of systemSourceParts){
 if(systemSourcePart === "IN.FIRE"){
  found = true;
 }
}
return found;
````
just make sure that each of your paths return a boolean.

Also sidenote the usage of item.system.source was just an example you can reference anything on item.
e.g. my potion detections for dnd5e
``
return item.type === "consumable" && item.system.consumableType === "potion"
``

**Comment by AngryBeaver** [comment]
to make it more robust and less fail prone you can/should make use of trim() or and toLowerCase().

so when you edit your item source, you can actuall do spaces or Upercases however you like:  In.Fire          | IN.Acid

**Reply by thangurion** [reply]
Holy sh... Works perfectly. You make my day.
